java/src/main/java/com/genexus/reports/PDFReportItext8.java

```diff
@@ -614,12 +635,37 @@ else if (valign == VerticalAlign.BOTTOM.value())
 				Rectangle htmlRectangle = new Rectangle(llx, lly, urx - llx, ury - lly);
 				YPosition yPosition = new YPosition(htmlRectangle.getTop());
 
-				ConverterProperties converterProperties = new ConverterProperties();
-				converterProperties.setFontProvider(document.getFontProvider());
+				Canvas htmlCanvas = new Canvas(cb, htmlRectangle);
+				htmlCanvas.setFontProvider(fontProvider);
+
 				//Iterate over the elements (a.k.a the parsed HTML string) and handle each case accordingly
 				List<IElement> elements = HtmlConverter.convertToElements(sTxt, converterProperties);
-				for (IElement element : elements)
-					processHTMLElement(htmlRectangle, yPosition, (IBlockElement) element);
+				for (IElement element : elements) {
+					float blockElementHeight = getBlockElementHeight((IBlockElement)element, htmlRectangle);
+					if (pageHeightExceeded(bottomMargin, yPosition.getCurrentYPosition()))
+					{
+						llx = leftAux + leftMargin;
+						lly = drawingPageHeight - bottomAux;
+						urx = rightAux + leftMargin;
+						ury = drawingPageHeight - topAux;
+						htmlRectangle = new Rectangle(llx, lly, urx - llx, ury - lly);
+						yPosition = new YPosition(htmlRectangle.getTop());
+						bottomAux -= drawingPageHeight;
+						GxEndPage();
+						GxStartPage();
+
+						cb = new PdfCanvas(pdfPage);
+						sTxt = sTxt.replaceAll("\\s+$", "");
+						cb.setFontAndSize(this.baseFont, fontSize);
+						cb.setFillColor(new DeviceRgb(foreColor));
+
+						htmlCanvas = new Canvas(cb, htmlRectangle);
```

Comment:
This line and the next one it seems to be the same that is in line 638.
May be it is good to write it in a method so if any time it's changed you will be sure to change it in both side.